### PR TITLE
feat: preserve message text in the input field

### PR DESCRIPTION
**Pull request by juliahermak**
[(after)Preserve message text in the input field.webm](https://github.com/Adamant-im/adamant-im/assets/122323015/d81bacf2-c49c-4ee9-8e2b-9ed027600b90)

### CODE REVIEW

File: src/components/AChat/AChatForm.vue [hunk 1]
```diff
@@ -119,7 +123,9 @@ export default {
   methods: {
     submitMessage() {
       this.$emit('message', this.message)
-      this.message = ''
+      if (this.clearInputValueOnSend === true) {
```
Comment (bludnic) [inline src/components/AChat/AChatForm.vue:126]: shorter variant
```suggestion
      if (this.clearInputValueOnSend) {
```

File: src/components/Chat/Chat.vue [hunk 2]
```diff
@@ -209,6 +209,7 @@
           :message-text="$route.query.messageText"
           @message="onMessage"
           @esc="replyMessageId = -1"
+          :clearInputValueOnSend="this.$store.state.balance < 0.001 === false"
```
Comment (bludnic) [inline src/components/Chat/Chat.vue:212]: compare other side with `>=` and use shorter variant

```suggestion
          :clearInputValueOnSend="this.$store.state.balance >= 0.001"
```

Comment (bludnic) [reply src/components/Chat/Chat.vue:212]: rename the prop name by using snake-case, just to keep the same style in the codebase

**Review by bludnic — CHANGES_REQUESTED** [review]

**Comment by bludnic** [comment]
[Feat: Preserve message text in the input field, if it was not sent](https://trello.com/c/HUvfBtph/363-feat-preserve-message-text-in-the-input-field-if-it-was-not-sent)

**Review by bludnic — APPROVED** [review]

**Review by adamant-al — CHANGES_REQUESTED** [review]
I don't like the custom-case solution.
We have `validateMessage` method.
Also, consider moving magic numbers to constants.
Refactor code, if appropriate.

**Comment by bludnic** [comment]
> I don't like the custom-case solution. We have `validateMessage` method. Also, consider moving magic numbers to constants. Refactor code, if appropriate.

I need to think about how to make the validation in one place and share it between Chat and AChatForm component, then I will came with a solution

**Comment by bludnic** [comment]
- Added `validator` prop for `AChatForm`
- Replaced harcoded fee from constants

@adamant-al

File: src/components/Chat/Chat.vue [hunk 3]
```diff
@@ -305,31 +307,33 @@ function getUserMeta(userId) {
 /**
  * Validate message before sending.
  * @param {string} message
- * @returns {boolean}
+ * @returns {string | false} If `false` then validation passed without errors.
  */
+const validationErrors = {
```
[the remark this answers is not in the document]
Comment (bludnic) [reply src/components/Chat/Chat.vue:307]: moved above the docblock

**Pull request merged**